packages/core/src/routes/connector.ts

```diff
@@ -59,6 +59,24 @@ export default function connectorRoutes<T extends AuthedRouter>(router: T) {
         'connector.more_than_one_sms'
       );
 
+      const targetAndPlatformObjectsInUse = connectors
+        .filter(({ type }) => type === ConnectorType.Social)
+        .map(({ metadata: { target, platform } }) => ({
+          target,
+          platform,
+        }));
+
+      for (const { platform, target } of targetAndPlatformObjectsInUse) {
+        if (platform === null) {
+          continue;
+        }
+        const filteredConnectors = connectors.filter(
+          ({ metadata: { target: connectorTarget, platform: connectorPlatform } }) =>
+            platform === connectorPlatform && target === connectorTarget
+        );
+        assertThat(filteredConnectors.length <= 1, 'connector.multiple_target_with_same_platform');
+      }
```

Comment:
should have a simpler way to check the dupliates in one single loop.

Reply:
Updated with a more efficient way.